MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -117,26 +129,27 @@ public void Logout (WebServiceLogin login)
 		}
 
 		[WebMethod]
-		public void CreateLanefile (WebServiceLogin login, int lane_id, string filename)
+		public void CreateLanefile (WebServiceLogin login, DBLane lane, string filename)
 		{
 			if (string.IsNullOrEmpty (filename))
 				throw new ArgumentException ("filename");
 
-			if (lane_id <= 0)
+			if (lane.id <= 0)
 				throw new ArgumentException ("lane_id");
 
 			using (DB db = new DB ()) {
-				VerifyUserInRole (db, login, Roles.Administrator);
-				db.Audit (login, "WebServices.CreateLaneFile (lane_id: {0}, filename: {1})", lane_id, filename);
+				VerifyUserInRoles (db, login, lane.required_roles, false);
```

Comment:
This is not good, here you're taking `required_roles` as input, which means anybody can make a request and set `required_roles` to themselves.

You need to fetch the required roles from the database here, not accept the input.